Development/nmos/system_resources.cpp

```diff
@@ -35,13 +69,32 @@ namespace nmos
         const auto& is04 = nmos::fields::is04(data);
         const auto& ptp = nmos::fields::ptp(data);
 
+        auto global_data = value_of({
+            { nmos::fields::registration_heartbeat_interval, nmos::fields::heartbeat_interval(is04) },
+            { nmos::fields::ptp_announce_receipt_timeout, nmos::fields::announce_receipt_timeout(ptp) },
+            { nmos::fields::ptp_domain_number, nmos::fields::domain_number(ptp) },
+            { nmos::fields::tags, data.at(nmos::fields::tags) }
+        });
+
+        if (data.has_field(nmos::fields::syslog))
```

Comment:
Were `nmos::fields::syslog` defined with `field_as_value_or` as suggested above, this could be written as:
```C++
const auto& syslog = nmos::fields::syslog(data);
if (!syslog.is_null())
```
This avoids finding the field twice.

Reply:
done